### PR TITLE
HL-251 | Fix - De Minimis grant has an initial selected value

### PR DESCRIPTION
**Pull request by Nizar-Rahme**
## Description :sparkles:
- Fix mismatching application types
- Remove the unnecessary application template query
- Fix save logic on the first step for new applications
- Fix Attachment types for shared, Benefit, and Kesäseteli modules

## Testing :alembic:
- Create a new application
- The question about the De Minimis grant should have no pre-selected value

**Note:** There was a bug when saving **new** applications using the "save and quit" button **on the first step**, this also got fixed in the same task.

### CODE REVIEW

File: frontend/benefit/applicant/src/hooks/useFormActions.tsx [hunk 1]
```diff
@@ -29,21 +29,19 @@ const useFormActions = (application: Application): FormActions => {
...
-    }
-  }, [newApplication?.id, router]);
+  const { mutateAsync: createApplication, error: createApplicationError } =
+    useCreateApplicationQuery();
+
+  const createApplicationAndAppendId = async (
+    data: ApplicationData
+  ): Promise<void> => {
+    const newApplication = await createApplication(data);
+    void router.replace({
```
Comment (nenonja) [inline frontend/benefit/applicant/src/hooks/useFormActions.tsx:39]: router replace causes page refresh, but you can prevent it by passing `{ shallow: true }`. I'm not sure though which is the use case here.

**Review by nenonja — CHANGES_REQUESTED** [review]
One minor comment... otherwise refactoring with attachments look good. Still kesaseteli component tests and deploying has failed, you should address them. New Browser test coverage some part of the attachments but does not try to add wrong type of file for example so it needs some manual testing for kesaseteli (I Can do it when deploy works).

File: frontend/benefit/applicant/src/hooks/useFormActions.tsx (hunk 1 above)
Comment (Nizar-Rahme) [reply frontend/benefit/applicant/src/hooks/useFormActions.tsx]: Good point 👍 
Fixed

**Comment by Nizar-Rahme** [comment]
> One minor comment... otherwise refactoring with attachments look good. Still kesaseteli component tests and deploying has failed, you should address them. New Browser test coverage some part of the attachments but does not try to add wrong type of file for example so it needs some manual testing for kesaseteli (I Can do it when deploy works).

I made TestCafe uses the same TS version that is used in the applications, which fixed the failure in building browser tests.

**Review by nenonja — APPROVED** [review]
It seems that everything is green now ✅

**Pull request merged**